extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/BeanArchivePredicateBuildItem.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.arc.deployment;
+
+import java.util.function.Predicate;
+
+import io.quarkus.builder.item.MultiBuildItem;
+import io.quarkus.deployment.ApplicationArchive;
+
+/**
+ * 
+ * By default, only explict/implicit bean archives (as defined by the spec) are considered during the bean discovery. However,
+ * extensions can register a logic to identify additional bean archives.
+ */
+public final class BeanArchivePredicateBuildItem extends MultiBuildItem {
```

Comment:
I'm not sure about the name of this build item. The other alternative was `AdditionalBeanArchiveBuildItem` but given the fact that the item does not represent a bean archive but a logic that identifies a bean archive I decided for `BeanArchivePredicateBuildItem`.

Reply:
I think `BeanArchivePredicateBuildItem` makes sense.